File: 2033-minimum-operations-to-make-a-uni-value-grid/2033-minimum-operations-to-make-a-uni-value-grid.js
/**
 * @param {number[][]} grid
 * @param {number} x
 * @return {number}
 */
var minOperations = function(grid, x) {
        grid = grid.flat().sort((a, b) => a-b)
        let target = grid[Math.floor(grid.length / 2)]
        let answer = 0
        for (let g of grid){
            if((target-g) % x == 0){
                answer += Math.abs(target - g) / x
            }
            else
                return -1
        }

        return answer
};

File: 2033-minimum-operations-to-make-a-uni-value-grid/2033-minimum-operations-to-make-a-uni-value-grid.py
class Solution:
    def minOperations(self, grid: List[List[int]], x: int) -> int:
        list2 = sum(grid, [])
        list2.sort()

        target = list2[len(list2)//2]
        answer = 0


        for value in list2:
            if abs(target-value) % x == 0:
               answer += abs(target-value) // x
            else:
                return -1
        return answer
    

